docs/source/plugins/developing_plugins.rst

```diff
@@ -2060,6 +2061,32 @@ subsequent sections.
             ctx.panel.set_state("event", "on_change_group_slice")
             ctx.panel.set_data("event_data", event)
 
+        def on_change_spaces(self, ctx):
+            """Implement this method to set panel state/data when the current
+            state of spaces changes in the app.
+
+            The current state of spaces will be available via ``ctx.spaces``.
+            """
+            event = {
+                "data": ctx.spaces,
+                "description": "the current state of spaces",
+            }
+            ctx.panel.set_state("event", "on_change_spaces")
+            ctx.panel.set_data("event_data", event)
+
+        def on_change_workspace(self, ctx):
```

Comment:
Potentially this should be deleted if you agree with my inclination to remove the `on_change_workspace` event.

If it remains, then I believe `ctx.workspace` here should now be `ctx.spaces.name`.